frame/staking/src/lib.rs

```diff
@@ -539,33 +611,50 @@ pub struct Nominations<AccountId> {
 
 /// The amount of exposure (to slashing) than an individual nominator has.
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug)]
-pub struct IndividualExposure<AccountId, Power: HasCompact> {
+pub struct IndividualExposure<AccountId, RingBalance, KtonBalance>
+where
+	RingBalance: HasCompact,
+	KtonBalance: HasCompact,
+{
 	/// The stash account of the nominator in question.
 	who: AccountId,
 	/// Amount of funds exposed.
 	#[codec(compact)]
-	value: Power,
+	ring_balance: RingBalance,
+	#[codec(compact)]
+	kton_balance: KtonBalance,
+	power: Power,
 }
 
 /// A snapshot of the stake backing a single validator in the system.
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default, RuntimeDebug)]
-pub struct Exposure<AccountId, Power: HasCompact> {
+pub struct Exposure<AccountId, RingBalance, KtonBalance>
+where
+	RingBalance: HasCompact,
+	KtonBalance: HasCompact,
+{
+	/// The validator's own stash that is exposed.
+	#[codec(compact)]
+	pub own_ring_balance: RingBalance,
+	#[codec(compact)]
+	pub own_kton_balance: KtonBalance,
+	pub own_power: Power,
```

Comment:
Yes.

Power is a dynamic value, we can't use it for slashing. So in previous version we use a perbill which calculated from `own_power/total_power` and apply it to active_ring/kton, but this will cause a problem which  was mentioned in #222 (if staker bond or unbond the active_ring/kton will changed immediately). We need a snapshot of ledger to perform the slashing at that moment. Also we should slash their unlocking items (if their active currency not enough for slashing).